exporter/collector/metricsexporter.go

```diff
@@ -97,23 +123,82 @@ func (me *metricsExporter) pushMetrics(ctx context.Context, m pdata.Metrics) err
 			mes := ilm.Metrics()
 			for k := 0; k < mes.Len(); k++ {
 				metric := mes.At(k)
+				for _, md := range me.mapper.metricDescriptor(metric) {
+					me.mds <- md
+				}
 				timeSeries = append(timeSeries, me.mapper.metricToTimeSeries(monitoredResource, extraLabels, metric)...)
 			}
 		}
 	}
 
 	// TODO: self observability
+	// TODO: Figure out how to configure service time series calls.
+	if false {
+		err := me.createServiceTimeSeries(ctx, timeSeries)
+		recordPointCount(ctx, len(timeSeries), m.DataPointCount()-len(timeSeries), err)
+		return err
+	}
 	err := me.createTimeSeries(ctx, timeSeries)
 	recordPointCount(ctx, len(timeSeries), m.DataPointCount()-len(timeSeries), err)
 	return err
 }
 
+// Reads metric descriptors from the md channel, and reports them (once) to GCM.
+func (me *metricsExporter) exportMetricDescriptorRunner() {
+	mdCache := make(map[string]*metricpb.MetricDescriptor)
+	// We iterate over all metric descritpors until the channel is closed.
+	// Note: if we get terminated, this will still attempt to export all descriptors
+	// prior to shutdown.
+	for md := range me.mds {
+		// Not yet sent, now we sent it.
+		// TODO - check to see if this is a service/system metric and doesn't send descriptors.
+		if !me.cfg.MetricConfig.SkipCreateMetricDescriptor && md != nil && mdCache[md.Type] == nil {
+			err := me.exportMetricDescriptor(context.Background(), md)
+			// TODO: Log-once on error, per metric descriptor?
+			// TODO: Re-use passed-in logger to exporter.
+			if err != nil {
+				log.Printf("Unable to send metric descriptor: %s, %s", md, err)
+			}
+			mdCache[md.Type] = md
```

Comment:
If you intend to retry the CMD next time around if there was an error the first go, there is a bug here

```suggestion
			if err != nil {
				log.Printf("Unable to send metric descriptor: %s, %s", md, err)
			} else {
				mdCache[md.Type] = md
			}
```

Reply:
Good catch!